packages/trader/src/AppV2/Containers/Trade/__tests__/trade-types.spec.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockStore } from '@deriv/stores';
+import TradeTypes from '../trade-types';
+import TraderProviders from '../../../../trader-providers';
+import { getTradeTypesList } from 'AppV2/Utils/trade-types-utils';
+
+jest.mock('AppV2/Utils/trade-types-utils');
+
+const mockGetTradeTypesList = getTradeTypesList as jest.MockedFunction<typeof getTradeTypesList>;
+
+const contract_types_list = {
+    rise_fall: {
+        name: 'Rise/Fall',
+        categories: [
+            { text: 'Rise', value: 'rise' },
+            { text: 'Fall', value: 'fall' },
+        ],
+    },
+    vanilla: {
+        name: 'Vanilla',
+        categories: [
+            { text: 'Vanilla Call', value: 'vanilla_call' },
+            { text: 'Vanilla Put', value: 'vanilla_put' },
+        ],
+    },
+};
+
+const default_mock_store = {
+    modules: {
+        trade: {
+            contract_type: 'rise_fall',
+            contract_types_list,
+            onMount: jest.fn(),
+            onUnmount: jest.fn(),
+        },
+    },
+};
+
+const mockTradeTypes = (mocked_store = mockStore(default_mock_store)) => {
+    return (
+        <TraderProviders store={mocked_store}>
+            <TradeTypes
+                onTradeTypeSelect={jest.fn()}
+                trade_types={mockGetTradeTypesList(default_mock_store.modules.trade.contract_types_list)}
+                contract_type='rise_fall'
+            />
+        </TraderProviders>
+    );
+};
+
+describe('TradeTypes', () => {
+    beforeEach(() => {
+        mockGetTradeTypesList.mockReturnValue([
+            { value: 'rise', text: 'Rise' },
+            { value: 'fall', text: 'Fall' },
+            { value: 'vanilla_call', text: 'Vanilla Call' },
+            { value: 'vanilla_put', text: 'Vanilla Put' },
+        ]);
+    });
+
+    it('should render the TradeTypes component with pinned and other trade types', () => {
+        render(mockTradeTypes());
+
+        expect(screen.getByText('View all')).toBeInTheDocument();
+        expect(screen.getByText('Rise')).toBeInTheDocument();
+    });
+
+    it('should open ActionSheet when View all button is clicked', async () => {
+        render(mockTradeTypes());
+
+        await userEvent.click(screen.getByText('View all'));
+
+        expect(screen.getByText('Trade types')).toBeInTheDocument();
+        expect(screen.getByText('Fall')).toBeInTheDocument();
+    });
+
+    it('should handle adding and removing pinned trade types', async () => {
+        render(mockTradeTypes());
+
+        await userEvent.click(screen.getByText('View all'));
```

Comment:
perhaps we could introduce a variable for 'View all' because it's checked 3 times?)

Reply:
It's just a test file 😄